src/Protocol/ActivityPub/Processor.php

```diff
@@ -903,20 +907,15 @@ public static function followUser($activity)
 		if (!empty($cid)) {
 			self::switchContact($cid);
 			DBA::update('contact', ['hub-verify' => $activity['id'], 'protocol' => Protocol::ACTIVITYPUB], ['id' => $cid]);
-			$contact = DBA::selectFirst('contact', [], ['id' => $cid, 'network' => Protocol::NATIVE_SUPPORT]);
-		} else {
-			$contact = [];
 		}
 
 		$item = ['author-id' => Contact::getIdForURL($activity['actor']),
 			'author-link' => $activity['actor']];
 
-		$note = Strings::escapeTags(trim($activity['content'] ?? ''));
-
 		// Ensure that the contact has got the right network type
 		self::switchContact($item['author-id']);
 
-		$result = Contact::addRelationship($owner, $contact, $item, false, $note);
+		$result = Contact::addRelationship($owner, [], $item, false, $activity['content'] ?? '');
```

Comment:
Why have you removed the contact array from here?

Reply:
Because it is also retrieved based on the `author-id` key of the `$item` in `Contact::addRelationship` and overwrites the `$contact` parameter, which made the retrieval in this method completely superfluous whether it failed or succeeded.

Reply:
The `author-id` is the public contact, that record is the personal contact. Although there is some other fallback in the code, I would like to keep that code in the RC phase since it isn't testable so easily.

Reply:
Here's the full rundown of how it was working (or not, sometimes) in this particular instance.

First, in `src/Protocol/ActivityPub/Processor.php`:
1. Line 892: `$uid` is obtained from the activity object ID which is the importer URL.
2. Line 997: `$owner` is obtained from `$uid`. It's the importer local user array.
3. Line 902: `$cid` is obtained from the activity actor URL and `$uid`. It's a contact ID specific to the importer.
4. Line 906: `$contact` is populated from this same `$cid`.
5. Line 911: `$item` is populated from the public contact ID from the activity actor URL.

Now we go over to `src/Model/Contact.php`:
1. Line 2386: At this point, `$importer` is the local user retrieved earlier, `$contact` has the user-specific contact data if it exists, and `$item` has the public contact ID.
2. Line 2412: `$cid` is populated with the contact ID specific to the importer user.
3. Line 2414: If this contact exists, `$contact` is then overridden with the user-specific contact data, which it already contained if it existed.

The logical conclusion is that the user-specific contact data retrieval in  `src/Protocol/ActivityPub/Processor.php:906` is superfluous because, in this specific case, the data we pass will be overridden with the same data in `src/Model/Contact.php:2414`. Of course I could have added a simple variable type check to `$contact` before passing it to `Contact::addRelationship`, but in this case the `$contact` variable isn't useful at all since it's overridden with the same exact data in the called `Contact` method. Now it's out of the question to change the general `Contact::addRelationship` method, but we can definitely change the way we call it in `Processor::followUser` without any side-effect.

Reply:
We really should change this function in the future since the parameters are really confusing.